### PR TITLE
Randomprime v1.23.0

### PR DESCRIPTION
**Pull request by toasterparty**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -105,6 +105,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Metroid Prime
 
+- Fixed: Crashes in several rooms when skipping cutscenes exactly 1 second from the end
```
Comment (Miepee) [inline CHANGELOG.md:108]: i thought this was 1 frame?

File: CHANGELOG.md [hunk 2]
```diff
@@ -105,6 +105,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fixed: Incorrect shield texture for vertical **PowerBeamOnly** doors
```
Comment (Miepee) [inline CHANGELOG.md:109]: RDV calls these `Power Beam Only Door`. No camel case.

File: CHANGELOG.md [hunk 3]
```diff
@@ -105,6 +105,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fixed: Mine Security Station - Wave Pirates not dropping down if the player didn't skip the cutscene immediately
+- Fixed: Uncrashed Frigate - Escape music sometimes playing at inappropriate times
+- Fixed: Furnace - Ghost elements re-appearing when re-entering from East Furnace Access
+- Fixed: Ruined Courtyard - Water platforms breaking when skipping cutscene on the last possible frame
+- Changed: Updated tournament winners scan
+- Changed: Ventilation Shaft: Cutscene skip no longer waits for load
+- Changed: Mine Security Station: Timing of activation of Wave Pirates is now the same regardless of what death animation the Shadow Pirates Play
+- Changed: Mine Security Station: Adjusted cutscene trigger so it can't be accidentally skipped
+- Changed: Ruined Shrine: Adjusted position of the cutscene skip lock-on point
+- Changed: Minor item acquisition jingle for health/pb/missile refill instead of major
```
Comment (Miepee) [inline CHANGELOG.md:122]: refills dont exist in rdv yet.

File: CHANGELOG.md [hunk 4]
```diff
@@ -105,6 +105,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Changed: Sun Tower Elevator: Rotated Samus Actor 90 degrees
```
Comment (Miepee) [inline CHANGELOG.md:123]: this reads confusing, the actor is only rotated during a cutscene i thought?

File: CHANGELOG.md [hunk 5]
```diff
@@ -105,6 +105,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fixed: Crashes in several rooms when skipping cutscenes exactly 1 second from the end
+- Fixed: Incorrect shield texture for vertical **PowerBeamOnly** doors
+- Fixed: Elite Research - Stutter when loading room
+- Fixed: Ruined Shrine Access - Still playing Beetle fight music after leaving
+- Fixed: Research Entrance - Soflock if player kills pirates before touching the cutscene trigger (reverse labs)
+- Fixed: Mine Security Station - Wave Pirates not dropping down if the player didn't skip the cutscene immediately
+- Fixed: Uncrashed Frigate - Escape music sometimes playing at inappropriate times
+- Fixed: Furnace - Ghost elements re-appearing when re-entering from East Furnace Access
+- Fixed: Ruined Courtyard - Water platforms breaking when skipping cutscene on the last possible frame
+- Changed: Updated tournament winners scan
```
Comment (Miepee) [inline CHANGELOG.md:117]: feel free to ignore, but it looks more visually pleasing if all the changed ones come before the fixed.

**Review by Miepee — COMMENTED** [review]
I am not miguel, but here are some comments.

File: CHANGELOG.md (hunk 1 above)
Comment (toasterparty) [reply CHANGELOG.md:108]: It's a frame-perfect start press 1 second before end

The fade-out takes 1 second to complete and it's when that fade ends that we have the edge case

File: CHANGELOG.md (hunk 5 above)
Comment (toasterparty) [reply CHANGELOG.md:117]: A really long time ago @henriquegemignani told me he prefers fixed first

Comment (toasterparty) [reply CHANGELOG.md:117]: Preferred was something like:

```
Fixed
Added
Changed
Removed
```

Comment (Miepee) [reply CHANGELOG.md:117]: looked at what keepachangelog recommends, which is this:
```
added
changed
deprecated
removed
fixed
security
```

Which looks worse to me. but also who really cares about the order.

Comment (toasterparty) [reply CHANGELOG.md:117]: let's both agree that anything is better than mixing/alternating types

**Comment by toasterparty** [comment]
> What about the changes from
> 
> [randovania/randomprime#25](https://github.com/randovania/randomprime/pull/25) and [randovania/randomprime#26](https://github.com/randovania/randomprime/pull/26)?

🌭

File: CHANGELOG.md [hunk 6]
```diff
@@ -105,6 +105,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+- Fixed: Crashes in several rooms when pressing start to skip cutscene exactly 1 second from the end of cutscene
+- Fixed: Incorrect shield texture for vertical **Power Beam Only** doors
+- Fixed: Elite Research - Stutter when loading room
+- Fixed: Ruined Shrine Access - Still playing Beetle fight music after leaving
+- Fixed: Research Entrance - Soflock if player kills pirates before touching the cutscene trigger (reverse labs)
+- Fixed: Mine Security Station - Wave Pirates not dropping down if the player didn't skip the cutscene immediately
+- Fixed: Uncrashed Frigate - Escape music sometimes playing at inappropriate times
+- Fixed: Furnace - Ghost elements re-appearing when re-entering from East Furnace Access
+- Fixed: Ruined Courtyard - Water platforms breaking when skipping cutscene on the last possible frame
```
Comment (MrMiguel211) [inline CHANGELOG.md:116]: Line 116
This one should not be included as it's a fix of a new feature that never got released, doesn't make much sense to me to put it in.

File: CHANGELOG.md (hunk 4 above)
Comment (MrMiguel211) [reply CHANGELOG.md]: Technically all of the cutscene elements were rotated relative to the Samus actor, but just mentioning the actor is simpler.

File: CHANGELOG.md [hunk 7]
```diff
@@ -105,7 +105,40 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Metroid Prime
 
-- Changed: The map tracker uses the same names for elevators as when editing a preset.
+- Fixed: Crashes in several rooms when pressing start to skip cutscene exactly 1 second from the end of cutscene
+- Fixed: Incorrect shield texture for vertical **Power Beam Only** doors
+- Fixed: Elite Research - Stutter when loading room
+- Fixed: Ruined Shrine Access - Still playing Beetle fight music after leaving
+- Fixed: Mine Security Station - Wave Pirates not dropping down if the player didn't skip the cutscene immediately
+- Fixed: Uncrashed Frigate - Escape music sometimes playing at inappropriate times
```
Comment (MrMiguel211) [inline CHANGELOG.md:113]: Line 113
Not playing at inappropriate times, just stopping then playing again when left Reactor Core after PQ death

File: CHANGELOG.md [hunk 8]
```diff
@@ -105,7 +105,40 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fixed: Furnace - Ghost elements re-appearing when re-entering from East Furnace Access
+- Fixed: Main Plaza - Vault Ledge door cosmetic fix
+- Fixed: Hive Totem - Incorrect water height when *not* skipping cutscene, improved cutscene skip timing
```
Comment (MrMiguel211) [inline CHANGELOG.md:116]: 116
Inaccurate, would be better to just leave it at "Improved intro Cutscene Skip" and maybe change it from Fix to Change.

File: CHANGELOG.md [hunk 9]
```diff
@@ -105,7 +105,40 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fixed: Control Tower - Flying Pirates incorrectly flying away from the player when skipping cutscene
+- Fixed: Research Core - Combat Visor being forced upon grabbing the pickup on Competitive Cutscene mode
+- Fixed: Research Entrance - Soflock if player kills pirates before touching the cutscene trigger (reverse labs)
+- Fixed: Research Entrance - Fog disappearing after clearing the 1st Pass Pirates and before the 2nd Pass Shadow Pirates
+- Fixed: Energy Core - Underwater sound incorrectly playing even if you are not underwater
+- Fixed: Energy Core - Puzzle music no longer incorrectly plays if player leaves room during countdown
+- Fixed: Ruined Shrine: Beetle music incorrectly continues playing after leaving via morph tunnel
```
Comment (MrMiguel211) [inline CHANGELOG.md:123]: 123
Leaving via bottom door*

File: CHANGELOG.md [hunk 10]
```diff
@@ -105,7 +105,40 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Changed: Mine Security Station: Adjusted cutscene trigger so it can't be accidentally skipped
+- Changed: Ruined Shrine: Adjusted position of the cutscene skip lock-on point
+- Changed: Sun Tower Elevator: [Cosmetic] Samus Actor turned 90 degrees in cutscene
+- Changed: Control Tower - "Doors Unlocked" HUD Memo in Control Tower once the fight is done on Competitive Cutscene mode
+- Changed: Ruined Fountain - Morph Ball can no longer get stuck at the bend in Spider Track
+- Changed: Ruined Courtyard - Cosmetic improvement to the lower water cutscene
+- Changed: Energy Core - Increased the size of the Load Trigger to Furnace Access
+- Changed: Hive Totem - Adjusted the Hive Totem scan position to match PAL
+- Changed: Observatory - Restored an unused particle effect for the projector
+- Changed: Observatory - The projector now loads activated instead of activating immediately upon load
```
Comment (MrMiguel211) [inline CHANGELOG.md:141]: 140 - 141
Maybe mark these last two as cosmetic?

File: CHANGELOG.md [hunk 11]
```diff
@@ -105,7 +105,40 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fixed: Uncrashed Frigate - Escape music sometimes playing at inappropriate times
+- Fixed: Furnace - Ghost elements re-appearing when re-entering from East Furnace Access
+- Fixed: Main Plaza - Vault Ledge door cosmetic fix
+- Fixed: Hive Totem - Incorrect water height when *not* skipping cutscene, improved cutscene skip timing
+- Fixed: Control Tower - Flying Pirates incorrectly flying away from the player when skipping cutscene
+- Fixed: Research Core - Combat Visor being forced upon grabbing the pickup on Competitive Cutscene mode
+- Fixed: Research Entrance - Soflock if player kills pirates before touching the cutscene trigger (reverse labs)
+- Fixed: Research Entrance - Fog disappearing after clearing the 1st Pass Pirates and before the 2nd Pass Shadow Pirates
+- Fixed: Energy Core - Underwater sound incorrectly playing even if you are not underwater
+- Fixed: Energy Core - Puzzle music no longer incorrectly plays if player leaves room during countdown
```
Comment (MrMiguel211) [inline CHANGELOG.md:122]: 122
Puzzle music not playing again if re-entered the room during countdown*

**Pull request merged**